### PR TITLE
Disable 'memsw' by default under pbs_cgroups hook.

### PR DESCRIPTION
**Pull request by bayucan**
#### Describe Bug or Feature

* PBS ships with the 'memsw' turned on (enabled=true) via the default pbs_cgroups hook config file.
* This is not a good out-of-the box default as most Linux distributions disable memsw accounting. A swapaccount=1 is needed to be added to the kernel command line and reboot the node.
* The noticeable result is in mom_logs, where "Permission denied" messages show up when writing to missing *memsw cgroup files.

#### Describe Your Change

* Updated the default pbs_cgroups hook config file with an "enabled=false" setting under memsw.
* Added a test case in pbs_cgroup hook ptl testsuite that specifically test the default pbs_cgroups hook config file.
* Slight improvements made to the testsuite to lessen timeouts on test_big_group_cpuset on slower system.
#### Attach Test and Valgrind Logs/Output

* [ptl.cgroup_default_config.no_memsw.FAIL.txt](https://github.com/openpbs/openpbs/files/4732203/ptl.cgroup_default_config.no_memsw.FAIL.txt)
* [ptl.cgroup_default_config.no_memsw.PASS.txt](https://github.com/openpbs/openpbs/files/4732204/ptl.cgroup_default_config.no_memsw.PASS.txt)
* [ptl.cgroups_hook.no_memsw.txt](https://github.com/openpbs/openpbs/files/4732207/ptl.cgroups_hook.no_memsw.txt)
* [ptl.cgroups_hook.with_memsw.txt](https://github.com/openpbs/openpbs/files/4732213/ptl.cgroups_hook.with_memsw.txt)

### CODE REVIEW

File: test/tests/functional/pbs_cgroups_hook.py [hunk 1]
```diff
@@ -4093,9 +4094,41 @@ def test_vnodepernuma_use_hyperthreads(self):
...
+        self.load_default_config()
+        # Reduce the noise in mom_logs for existence=False matching
+        c = {'$logevent': '511'}
+        self.mom.add_config(c)
+        a = {'Resource_List.select': 'ncpus=1:mem=100mb'}
+        j = Job(TEST_USER, attrs=a)
+        j.create_script(self.sleep15_job)
+        jid = self.server.submit(j)
+        a = {'job_state': 'R'}
+        self.server.expect(JOB, a, jid)
```
Comment (bhroam) [inline test/tests/functional/pbs_cgroups_hook.py:4114]: You can combine the above two lines if you want.  They will fit <79 chars

File: test/tests/functional/pbs_cgroups_hook.py [hunk 2]
```diff
@@ -4093,9 +4094,41 @@ def test_vnodepernuma_use_hyperthreads(self):
...
+        err_msg = "write_value: Permission denied.*%s.*memsw" % (jid,)
```
Comment (bhroam) [inline test/tests/functional/pbs_cgroups_hook.py:4115]: I'd remove the comma from the tuple.  It doesn't need to be a tuple.

File: test/tests/functional/pbs_cgroups_hook.py [hunk 3]
```diff
@@ -4093,9 +4094,41 @@ def test_vnodepernuma_use_hyperthreads(self):
...
+        self.mom.log_match(err_msg, max_attempts=3, interval=1, n=100,
+                           regexp=True, existence=False)
+        self.server.status(JOB, ['exec_host'], jid)
+        ehost = j.attributes['exec_host']
+        ehost1 = ehost.split('/')[0]
+        ehjd1 = self.get_cgroup_job_dir('cpuset', jid, ehost1)
+        self.assertTrue(self.is_dir(ehjd1, ehost1))
```
Comment (bhroam) [inline test/tests/functional/pbs_cgroups_hook.py:4122]: The second argument to assertTrue() is a message.  That is just a host.  You should turn that into a real message.

File: test/tests/functional/pbs_cgroups_hook.py (hunk 1 above)
Comment (bayucan) [reply test/tests/functional/pbs_cgroups_hook.py:4114]: Good point.

File: test/tests/functional/pbs_cgroups_hook.py (hunk 2 above)
Comment (bayucan) [reply test/tests/functional/pbs_cgroups_hook.py:4115]: Ok.

File: test/tests/functional/pbs_cgroups_hook.py (hunk 3 above)
Comment (bayucan) [reply test/tests/functional/pbs_cgroups_hook.py:4122]: Ok, I'll add a message.

File: test/tests/functional/pbs_cgroups_hook.py [hunk 4]
```diff
@@ -4110,16 +4110,15 @@ def test_cgroup_default_config(self):
...
         self.mom.log_match(err_msg, max_attempts=3, interval=1, n=100,
                            regexp=True, existence=False)
         self.server.status(JOB, ['exec_host'], jid)
         ehost = j.attributes['exec_host']
         ehost1 = ehost.split('/')[0]
         ehjd1 = self.get_cgroup_job_dir('cpuset', jid, ehost1)
-        self.assertTrue(self.is_dir(ehjd1, ehost1))
+        self.assertTrue(self.is_dir(ehjd1, ehost1), "job cpuset dir not found")
         self.server.delete(id=jid, wait=True)
         self.assertFalse(self.is_dir(ehjd1, ehost1))
```
Comment (bhroam) [inline test/tests/functional/pbs_cgroups_hook.py:4123]: Do you want to add a message here as well?

Comment (bayucan) [reply test/tests/functional/pbs_cgroups_hook.py:4123]: Sure, I can add one here too.

**Comment by bayucan** [comment]
@bhroam : This should be ready to be merged...

**Comment by bayucan** [comment]
@bhroam : thanks!

**Pull request merged**